src/test/fuzz/util.cpp

```diff
@@ -14,6 +14,19 @@
 
 #include <memory>
 
+std::vector<uint8_t> ConstructPubKeyBytes(FuzzedDataProvider& fuzzed_data_provider, Span<const uint8_t> byte_data, const bool compressed) noexcept
+{
+    uint8_t pk_type;
+    if (compressed) {
+        pk_type = fuzzed_data_provider.PickValueInArray({0x02, 0x03});
+    } else {
+        pk_type = fuzzed_data_provider.PickValueInArray({0x04, 0x07});
```

Comment:
yeah, reading again more carefully, the code before was doing `ConsumeIntegralInRange(4, 7)`, which would have included 5 and 6, where 5 is nonsense in this context.

So probably fine to just squash the two commits and mention that it's not strictly a refactor because we are removing a nonsense value from the input.

Reply:
Removing nonsense in the first commit is fine. But I still prefer the keep the second commit seperate.

Reply:
Updated the comment!